internal/jobcontainers/process.go

```diff
@@ -0,0 +1,179 @@
+package jobcontainers
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"os/exec"
+	"sync"
+
+	"github.com/Microsoft/hcsshim/internal/cow"
+	"github.com/Microsoft/hcsshim/internal/log"
+	"golang.org/x/sys/windows"
+)
+
+// JobProcess represents a process run in a job object.
+type JobProcess struct {
+	cmd            *exec.Cmd
+	procLock       sync.Mutex
+	stdioLock      sync.Mutex
+	stdin          io.WriteCloser
+	stdout         io.ReadCloser
+	stderr         io.ReadCloser
+	waitBlock      chan struct{}
+	closedWaitOnce sync.Once
+	waitError      error
+}
+
+var _ cow.Process = &JobProcess{}
+
+func newProcess(cmd *exec.Cmd) *JobProcess {
+	return &JobProcess{
+		cmd:       cmd,
+		waitBlock: make(chan struct{}),
+	}
+}
+
+// ResizeConsole - TODO (dcantah): Implement console support with new pseudo console api
```

Comment:
I would prefer we file tracking items instead of (or at least in addition to) TODOs in code. TODOs are almost always forgotten.

Reply:
I was going to leave this, as it'll be removed after the os/exec fork PR and jobcontainers PR (after this one) are merged.